### PR TITLE
[binary addons] linux: use the same install paths as kodi core

### PR DESCRIPTION
**Pull request by wsnipex**
I've noticed that having to correct set install prefix of binary addons to the same as kodi is confusing to many users if you are not using default /usr/local on linux standalone builds.

This PR changes default install paths for bin addons to automatically be the same as kodi core, by adding the kodi prefix to kodi-config.cmake during kodi configure. Behavior for unified deps builds should be unchanged, but I admit I didn't thoroughly test that yet. 
While at it, I've tried to get rid of hardcoded "kodi" paths as much as possible to make rebranding easier, but there might be some spots left.

Finally, this also fixes a major bug with install paths on linux standalone builds I introduced in https://github.com/xbmc/xbmc/pull/6897 : 
we currently install addon libs to $prefix/lib/<cpu_arch>/kodi/addons, which is wrong since kodi installs to $prefix/lib/kodi/

### CODE REVIEW

File: project/cmake/scripts/common/prepare-env.cmake [hunk 1]
```diff
@@ -55,15 +56,15 @@ get_filename_component(prefix "${DEPENDS_PATH}" ABSOLUTE)
 set(APP_LIBDIR "${prefix}/lib")
```
Comment (Montellese) [inline project/cmake/scripts/common/prepare-env.cmake]: These variables can go since they don't seem to be used anymore in `kodi-config.cmake.in`.

**Comment by wsnipex** [comment]
jenkins build addons please

Btw, the drawback of this multi-arch fix is that pvr-addon debian packaging will be broken again :/
I'll take care of that once this PR is in.

**Comment by wsnipex** [comment]
@Montellese ok for you?

**Comment by Montellese** [comment]
Should be ok. Haven't tested it myself yet on win32 but jenkins seems to be happy.

**Comment by uNiversaI** [comment]
Pulling this patch over master at the when kodi finishes compiling, this message shows
```
install: cannot stat ‘project/cmake/scripts/linux/UseMultiArch.cmake’: No such file or directory
```

**Comment by wsnipex** [comment]
indeed, should be fixed now. Thanks for testing

**Comment by wsnipex** [comment]
jenkins build this please

**Comment by uNiversaI** [comment]
This is odd, you have

in  ```/usr/lib/kodi/addons/```
```
library.kodi.guilib  library.xbmc.codec  screensaver.rsxs.euphoria  screensaver.rsxs.solarwinds  visualization.glspectrum
library.xbmc.addon   library.xbmc.pvr  screensaver.rsxs.plasma  visualization.fishbmc  visualization.waveform
```
in ```/usr/lib/x86_64-linux-gnu/kodi/addons/ ``` you have 
```
audioencoder.flac  audioencoder.lame  audioencoder.vorbis  audioencoder.wav  visualization.shadertoy
```

Even after this pull ***In order for shadertoy to work***, I still need to copy manually

```/usr/lib/x86_64-linux-gnu/kodi/addons/visualization.shadertoy``` to ```/usr/lib/kodi/addons/visualization.shadertoy```

Wasnt this PR suppose to solve that?

**Comment by wsnipex** [comment]
@uNiversaI you must have stale files around. There is no way that anything is installing to /usr/lib/x86_64-linux-gnu with this PR. I've just re-verified that. But I've found a bug that caused addons to install to /usr/lib/kodi/kodi/ in some situations. This should be fixed now.
@Memphiz could you please check that OSX still installs to correct paths?

jenkins build this please

**Comment by wsnipex** [comment]
jenkins build addons please

**Comment by uNiversaI** [comment]
> @uNiversaI you must have stale files around. There is no way that anything is installing to /usr/lib/x86_64-linux-gnu with this PR.

This isnt for the addon itself its the .so files

I turned off ccache, git clean -xfd etc and in the case of shadertoy was a new fresh git clone only the so file installs to  /usr/lib/x86_64-linux-gnu.

The audio encoders same git clean -xfd before pulling from master and compiling so they are definitely installing to their so files as well to /usr/lib/x86_64-linux-gnu.

hint where to look for this *stale* files then pls.

**Comment by wsnipex** [comment]
make sure the addons are finding the correct kodi-config.cmake
How do you build?

**Comment by uNiversaI** [comment]
Shadertoy I build like.
  ```
  git clone https://github.com/topfs2/visualization.shadertoy
  cmake -DCMAKE_INSTALL_PREFIX=/usr
  sudo make
  sudo make install
  ```
Its interesting this morning it actually printed the correct install path, so clearly after booting system seems correct !?!
```
-- Installing: /usr/lib/kodi/addons/visualization.shadertoy/visualization.shadertoy.so.15.0
-- Installing: /usr/lib/kodi/addons/visualization.shadertoy/visualization.shadertoy.so
-- Installing: /usr/lib/kodi/addons/visualization.shadertoy/visualization.shadertoy.so.0.0.1
````
So ive **deleted** the ```/usr/lib/x86_64-linux-gnu/kodi``` 

now built audioencoders (I think I know slight pebcak I forgot to pull this patch when compiling audioencoders but usually built like.

    ```
  git clean -xffd && git reset --hard && git pull --rebase origin master
    sudo make -C tools/depends/target/xbmc-audioencoder-addons ADDONS="audioencoder.flac audioencoder.lame audioencoder.vorbis audioencoder.wav" PREFIX=/usr
    ```

Aftre pulling patch and recompiling audioencoders.
```
-- Install configuration: "Release"
-- Installing: /usr/lib/kodi/addons/audioencoder.wav/audioencoder.wav.so.1.0.0
-- Installing: /usr/lib/kodi/addons/audioencoder.wav/audioencoder.wav.so.15.0
-- Installing: /usr/lib/kodi/addons/audioencoder.wav/audioencoder.wav.so
-- Installing: /usr/share/kodi/addons/audioencoder.wav
-- Installing: /usr/share/kodi/addons/audioencoder.wav/addon.xml
-- Installing: /usr/share/kodi/addons/audioencoder.wav/icon.png
````

However flac lame ogg all failed to build smae reason

http://pastebin.com/A5sAAWAx

So I presume the same fix that went in for wav encoder needs to go into the remainder audio encoders as well.

**Comment by uNiversaI** [comment]
In fact I just realised something.

The audio encoders are ***probably*** failing to rip probably due to this very issue of being installed to /usr/lib/x86_64-linux-gnu/kodi/addons/

At some stage in not so distant past the install path was correct /usr/lib/kodi/addons/

I found out that when comparing the install dates between /x86_64-linux-gnu/kodi/addons/ 
audio encoders and /usr/lib/kodi there were a few months difference

So the files that were in /usr/lib/kodi/addons/ were months older and kodi reads those and not the ones in /usr/lib/x86_64-linux-gnu/kodi/addons/

I havent tested that theory but I presume this was causing more issues then previously realised.

**Comment by Memphiz** [comment]
@wsnipex still good on osx/ios

**Pull request merged**